### ISSUE TITLE
Sound stopped working after update / No audio with Dummy Output only

### ISSUE THREAD

**Issue by spacegears** [pop-os/pop#2584]
**Distribution (run `cat /etc/os-release`):**
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
`apt policy linux-firmware` output:
```
linux-firmware:
  Installed: 20220329.git681281e4-0ubuntu3.2+system76~1654629645~22.04~009c040
  Candidate: 20220329.git681281e4-0ubuntu3.2+system76~1654629645~22.04~009c040
  Version table:
     20220329.git681281e4-0ubuntu3.4 500
        500 http://apt.pop-os.org/ubuntu jammy-updates/main amd64 Packages
        500 http://apt.pop-os.org/ubuntu jammy-updates/main i386 Packages
 *** 20220329.git681281e4-0ubuntu3.2+system76~1654629645~22.04~009c040 1001
       1001 http://apt.pop-os.org/release jammy/main amd64 Packages
       1001 http://apt.pop-os.org/release jammy/main i386 Packages
        100 /var/lib/dpkg/status
     20220329.git681281e4-0ubuntu1 500
        500 http://apt.pop-os.org/ubuntu jammy/main amd64 Packages
        500 http://apt.pop-os.org/ubuntu jammy/main i386 Packages
```

**Issue/Bug Description:**
I am running an ASRock X300 with the 3.5mm speaker output, everything was working just fine until yesterday, when I installed some updates (maybe some Pipewire updates, too???) through Pop Shop, same as hundreds of times before. Then I shut down the computer after, turned it on later and I got no sound/audio ever since. I have tried everything I could think of with no luck. I am not a Linux Power User. I did look up the issue though and got nowhere with the possible solutions and fixes.

Settings shows `Dummy Output` as the only available Output Device.
Does anyone have any ideas? Any help is appreciated.

Here are some outputs of the commands I could find in another posts/issues:

`cat /proc/asound/cards` output:
`--- no soundcards ---`

`aplay --list-devices` output:
`aplay: device_list:275: no soundcards found...`

`lspci -v | grep -A7 -i "audio"` output:
```
02:00.1 Audio device: Advanced Micro Devices, Inc. [AMD/ATI] Raven/Raven2/Fenghuang HDMI/DP Audio Controller
	Subsystem: Advanced Micro Devices, Inc. [AMD/ATI] Raven/Raven2/Fenghuang HDMI/DP Audio Controller
	Flags: fast devsel, IRQ 255, IOMMU group 1
	Memory at fcd88000 (32-bit, non-prefetchable) [disabled] [size=16K]
	Capabilities: <access denied>
	Kernel modules: snd_hda_intel

02:00.2 Encryption controller: Advanced Micro Devices, Inc. [AMD] Family 17h (Models 10h-1fh) Platform Security Processor
	Subsystem: Advanced Micro Devices, Inc. [AMD] Family 17h (Models 10h-1fh) Platform Security Processor
--
02:00.6 Audio device: Advanced Micro Devices, Inc. [AMD] Family 17h (Models 10h-1fh) HD Audio Controller
	Subsystem: ASRock Incorporation Family 17h (Models 10h-1fh) HD Audio Controller
	Flags: fast devsel, IRQ 255, IOMMU group 1
	Memory at fcd80000 (32-bit, non-prefetchable) [disabled] [size=32K]
	Capabilities: <access denied>
	Kernel modules: snd_hda_intel

03:00.0 SATA controller: Advanced Micro Devices, Inc. [AMD] FCH SATA Controller [AHCI mode] (rev 61) (prog-if 01 [AHCI 1.0])
	Subsystem: ASRock Incorporation FCH SATA Controller [AHCI mode]
```
**^ is that `Capabilities: <access denied>` part normal?**

`lsmod | grep snd_hda_intel` output:
```
snd_hda_intel          53248  0
snd_intel_dspcfg       36864  1 snd_hda_intel
snd_hda_codec         167936  1 snd_hda_intel
snd_hda_core          114688  2 snd_hda_intel,snd_hda_codec
snd_pcm               151552  3 snd_hda_intel,snd_hda_codec,snd_hda_core
snd                   110592  10 snd_seq,snd_seq_device,snd_hwdep,snd_hda_intel,snd_hda_codec,snd_timer,snd_pcm,snd_rawmidi
```

`aplay -lL` output:
```
null
    Discard all samples (playback) or generate zero samples (capture)
samplerate
    Rate Converter Plugin Using Samplerate Library
speexrate
    Rate Converter Plugin Using Speex Resampler
jack
    JACK Audio Connection Kit
oss
    Open Sound System
pipewire
    PipeWire Sound Server
pulse
    PulseAudio Sound Server
upmix
    Plugin for channel upmix (4,6,8)
vdownmix
    Plugin for channel downmix (stereo) with a simple spacialization
default
    Default ALSA Output (currently PulseAudio Sound Server)
aplay: device_list:275: no soundcards found...
```

**Comment by spacegears** [comment]
So I just went through a dozen similar threads, no joy. The only difference I can see is that the drivers seem to be not loaded in my case, see below.

`inxi -SMA` output:
```
System:
  Host: x300-pop-os Kernel: 5.19.0-76051900-generic x86_64 bits: 64
    Desktop: GNOME 42.3.1 Distro: Pop!_OS 22.04 LTS
Machine:
  Type: Desktop Mobo: ASRock model: X300M-STX serial: <superuser required>
    UEFI: American Megatrends v: P1.40 date: 08/04/2020
Audio:
  Device-1: AMD Raven/Raven2/Fenghuang HDMI/DP Audio driver: N/A
  Device-2: AMD Family 17h HD Audio driver: N/A
  Sound Server-1: ALSA v: k5.19.0-76051900-generic running: yes
  Sound Server-2: PipeWire v: 0.3.56 running: yes
```

**Comment by max-pod** [comment]
Having similar problems that occur after my computer suspends / starts, and keep having to fix it with `sudo systemctl restart gdm `

**Comment by spacegears** [comment]
> Having similar problems that occur after my computer suspends / starts, and keep having to fix it with `sudo systemctl restart gdm `

Well, unfortunately it did not fix the issue in my case.

**Comment by lukmuk** [comment]
I can confirm having the exact same issue on my Asrock X300. Sound completely stopped working (only showing "Dummy output"), probably after a Pop-Shop upgrade.

Sound works when booting a current Pop!_OS image from a live USB stick.
Tested "Refresh OS" from preferences, but did not help. Maybe a clean installation is the only solution?

**Comment by lukmuk** [comment]
I did some further testing with installing Fedora 36. Here the sound is also gone (Dummy output) when updating the kernel to the latest version of `5.19.6` (the default upgrade after a clean installation, this will break the sound similar to the Pop!_OS upgrade), but it works for kernel `5.17.5` (the kernel shipped with the Fedora 36 image).

Since `pipewire` seems to depend on a working `alsa` (?), I suspected that a `alsa` update maybe the cause of the problem. 

On my Fedora installation with the latest kernel `5.19.6`, I downgraded `alsa` from `1.2.7` to version `1.2.6` and also did a downgrade of `pipewire`. The downgrades of the packages had **no** influence, resulting in a `Dummy Output` and no working sound.  
Downgrading the kernel helped: Sound works for me with updated `alsa` (`1.2.7`) and `pipewire` (`0.3.56.`) on kernel `5.17.5`.

**Working sound** on Fedora with kernel `5.17.5` (should be similar to working sound on kernel `5.18.10` on Pop!_OS):
alsa:
```
dnf list installed | grep alsa
alsa-lib.x86_64                                      1.2.7.2-1.fc36                      @updates     
alsa-sof-firmware.noarch                             2.1.1-1.fc36                        @updates     
alsa-ucm.noarch                                      1.2.7.2-1.fc36                      @updates     
alsa-utils.x86_64                                    1.2.7-1.fc36                        @updates     
pipewire-alsa.x86_64                                 0.3.56-1.fc36                       @updates     
qemu-audio-alsa.x86_64                               2:6.2.0-14.fc36                     @updates  
```

Pipewire:
```
dnf list installed | grep pipewire
pipewire.x86_64                                      0.3.56-1.fc36                       @updates     
pipewire-alsa.x86_64                                 0.3.56-1.fc36                       @updates     
pipewire-gstreamer.x86_64                            0.3.56-1.fc36                       @updates     
pipewire-jack-audio-connection-kit.x86_64            0.3.56-1.fc36                       @updates     
pipewire-libs.x86_64                                 0.3.56-1.fc36                       @updates     
pipewire-pulseaudio.x86_64                           0.3.56-1.fc36                       @updates     
pipewire-utils.x86_64                                0.3.56-1.fc36                       @updates 
```
Interesting: Output of `lspci -v | grep -A7 -i "audio"` shows an additional line when audio is working:  
**Kernel driver in use: snd_hda_intel**  
This is different from @spacegears result when audio is not working.

```
lspci -v | grep -A7 -i "audio"
04:00.1 Audio device: Advanced Micro Devices, Inc. [AMD/ATI] Raven/Raven2/Fenghuang HDMI/DP Audio Controller
	Subsystem: Advanced Micro Devices, Inc. [AMD/ATI] Raven/Raven2/Fenghuang HDMI/DP Audio Controller
	Flags: bus master, fast devsel, latency 0, IRQ 73, IOMMU group 1
	Memory at fcc88000 (32-bit, non-prefetchable) [size=16K]
	Capabilities: <access denied>
	Kernel driver in use: snd_hda_intel
	Kernel modules: snd_hda_intel

04:00.2 Encryption controller: Advanced Micro Devices, Inc. [AMD] Family 17h (Models 10h-1fh) Platform Security Processor
--
04:00.6 Audio device: Advanced Micro Devices, Inc. [AMD] Family 17h/19h HD Audio Controller
	Subsystem: ASRock Incorporation Device 2233
	Flags: bus master, fast devsel, latency 0, IRQ 52, IOMMU group 1
	Memory at fcc80000 (32-bit, non-prefetchable) [size=32K]
	Capabilities: <access denied>
	Kernel driver in use: snd_hda_intel
	Kernel modules: snd_hda_intel
```

Maybe kernel $\geq 5.19.0$ are not correctly loading up the snd_hda_intel driver for our hardware? 

I tried some modifications to `/etc/modprobe.d/alsa-base.conf`, as suggested on https://support.system76.com/articles/audio/, but without luck. Maybe there is a set of parameters for this configuration so that the kernel driver gets loaded correctly?

TL:DR: With the ASRock X300, downgrade to a kernel **below** `5.19.0`. This fixed sound issues for me. The problem persists at least to kernel `5.19.6`.

**Comment by spacegears** [comment]
I am happy to see that I am not the only one with this issue and that it is being mentioned on different forums, hopefully it will get resolved soon.

In the meantime, I just successfully downgraded to `5.18.19` and I can confirm that SOUND IS BACK NOW.

Shout out to **weredev** in this Reddit thread:
[https://www.reddit.com/r/pop_os/comments/jorxsc/best_way_to_revert_to_54_kernel/](https://www.reddit.com/r/pop_os/comments/jorxsc/best_way_to_revert_to_54_kernel/)

What I did is:
- Installed 5.18.19 using Mainline (you can get it from the Pop Shop)
- Booted into the old kernel (hold space during boot and select the 2nd option)
- Uninstalled 5.19.5
- Used the following command: `update-initramfs -k all -u`
- Booted into the old kernel again (because I had a newer one - 5.19.0 - installed)

Thank you for your work on this as well, @lukmuk.

**Comment by shreyes5** [comment]
Hi! Is there a fix for this issue using the latest kernel version? My outputs are pretty much the same as @spacegears and my kernel version is: 5.19.0-76051900-generic. Does someone know a fix for the dummy output issue?

**Comment by lukmuk** [comment]
> Hi! Is there a fix for this issue using the latest kernel version? My outputs are pretty much the same as @spacegears and my kernel version is: 5.19.0-76051900-generic. Does someone know a fix for the dummy output issue?

As suggested in this [Reddit comment](https://www.reddit.com/r/Fedora/comments/x4kh5z/comment/in0ostf/?utm_source=share&utm_medium=web2x&context=3) and this thread in the [OpenSUSE Bugzilla ](https://bugzilla.suse.com/show_bug.cgi?id=1202492) thread (starting from comment number 35), you can try to play with the amd_iommu=off or iommu=off as kernel parameters. Maybe this helps!

**Comment by shreyes5** [comment]
> > Hi! Is there a fix for this issue using the latest kernel version? My outputs are pretty much the same as @spacegears and my kernel version is: 5.19.0-76051900-generic. Does someone know a fix for the dummy output issue?
> 
> As suggested in this [Reddit comment](https://www.reddit.com/r/Fedora/comments/x4kh5z/comment/in0ostf/?utm_source=share&utm_medium=web2x&context=3) and this thread in the [OpenSUSE Bugzilla ](https://bugzilla.suse.com/show_bug.cgi?id=1202492) thread (starting from comment number 35), you can try to play with the amd_iommu=off or iommu=off as kernel parameters. Maybe this helps!

Alright, I'll try this and hope for the best.

**Comment by shreyes5** [comment]
> > Hi! Is there a fix for this issue using the latest kernel version? My outputs are pretty much the same as @spacegears and my kernel version is: 5.19.0-76051900-generic. Does someone know a fix for the dummy output issue?
> 
> As suggested in this [Reddit comment](https://www.reddit.com/r/Fedora/comments/x4kh5z/comment/in0ostf/?utm_source=share&utm_medium=web2x&context=3) and this thread in the [OpenSUSE Bugzilla ](https://bugzilla.suse.com/show_bug.cgi?id=1202492) thread (starting from comment number 35), you can try to play with the amd_iommu=off or iommu=off as kernel parameters. Maybe this helps!

I think iommu=off might lead me to some other difficulties. Can someone let me know as to how do I downgrade my kernel? I am currently on 5.19.0-76051900-generic, I'd like to downgrade to 5.15.x but I don't know how to go about doing that. If someone can help, that'll be great.

**Comment by mtzirkel** [comment]
Downgrading fixed my "Dummy Sound" issue as well. Thank you all

**Comment by loafylemon** [comment]
I just had booted to no sound issue this morning.

For some unknown reason, PipeWire was disabled and the system was looking for PulseAudio which does not exist on 22.04. Enabling PipeWire Daemon fixed it for me.

`systemctl --user --now enable pipewire pipewire-pulse`

What's peculiar is that this is a completely fresh installation, here's the list of things I did in the terminal the day before

```sudo apt update
sudo apt upgrade
flatpak update```

That's all.

**Comment by spacegears** [comment]
Fixed in the `6.0.0` kernel. Yay!

**Comment by emad-ramlawi** [comment]
Faced this same exact issue and had to go down kernel to resolve

currently using this kernel which has good audio:
`Linux pop-os 6.0.2-76060002-generic #202210150739~1666289067~22.04~fe0ce53 SMP PREEMPT_DYNAMIC Thu O x86_64 x86_64 x86_64 GNU/Linux`

And I have this laptop and i think the issue is with 
`ii  linux-image-6.0.3-76060003-generic      6.0.3-76060003.202210211149~1666452039~22.04~1891946            amd64        Linux kernel image for version 6.0.3 on 64 bit x86 SMP`


```
System Information
	Manufacturer: LENOVO
	Product Name: 20U9001NUS
	Version: ThinkPad X1 Carbon Gen 8
	Serial Number: MJ0EBB9R
	UUID: 23df734c-2fa5-11b2-a85c-8daab0eb162d
	Wake-up Type: Power Switch
	SKU Number: LENOVO_MT_20U9_BU_Think_FM_ThinkPad X1 Carbon Gen 8
	Family: ThinkPad X1 Carbon Gen 8
```

**Comment by Jemy09** [comment]
I can confirm having the very same issue with 
`Linux pop-os 6.0.3-76060003-generic #202210211149~1666452039~22.04~1891946`
Sound settings just say "dummy output"

pop-os                      
    description: Notebook
    product: Dell G15 5510 (0A64)
    vendor: Dell Inc.
    serial: 9DPTLF3
    width: 64 bits
    capabilities: smbios-3.2.0 dmi-3.2.0 smp vsyscall32
    configuration: boot=normal chassis=notebook family=GSeries

**Comment by skhalsa10** [comment]
I am having the same issue with `Linux 6.0.6-76060006-generic x86_64
`

**Comment by JGodin-C2C** [comment]
Oï , 
This solution is working , thanks @skhalsa10 , i guess the problem comes from the driver ?

**Comment by rmflight** [comment]
Hey @skhalsa10, thank you for that note!

On tuesday I got an updated kernel (6.2.6) and alsa, and the only sound that worked after wake from sleep was Bluetooth. Your solution finally fixed my sound issues and all my sound hardware is properly recognized again.

**Comment by skhalsa10** [comment]
yay glad it is all working. yeah normally hardware issues are drivers or bad hardware.
